packages/buttons/src/styled/StyledAnchor.ts

```diff
@@ -24,7 +25,7 @@ export const StyledAnchor = styled(StyledButton).attrs(props => ({
 }))`
   direction: ${props => props.theme.rtl && 'rtl'};
 
-  ${props => retrieveComponentStyles(COMPONENT_ID, props)};
+  ${props => retrieveComponentStyles(props['data-garden-id'], props)};
```

Comment:
`props` always include the ones set in `.attrs`. If we pursue with this type of migration for every component and drop the `componentId` param from `retrieveComponentStyles, we could rewrite this to:

```suggestion
  ${retrieveComponentStyles};
```

Reply:
I would discourage the rewrite at this point since a v10 goal is to consolidate packages. At that point, we should visit styled component re-wrapping in earnest and make sure we have the right solution for CSS [targeting](https://garden.zendesk.com/components/theme-provider#targeting).